spot_rl_experiments/spot_rl/envs/skill_manager.py

```diff
@@ -95,17 +98,24 @@ def nav(self, nav_target: str = None) -> Tuple[bool, str]:
             result = self.nav_controller.execute(nav_target_list)
         except Exception:
             return False, "Error encountered while navigating"
-        print(result)
-        # Check for success and return appropriately
-        # TODO: Please check if this is correct formulation of success.
-        # if is_pose_within_bounds(
-        #     result[0][-1].get("pose"),
-        #     nav_target_list[0],
-        #     self.nav_config.SUCCESS_DISTANCE,
-        #     self.nav_config.SUCCESS_ANGLE_DIST,
-        # ):
-        if True:
-            return True, "Successfully reached the target pose"
+
+        if self._should_record_trajectories:
+            _nav_target_pose = nav_target_list[0]
+            # Make the angle from rad to deg
+            _nav_target_pose_deg = (
+                _nav_target_pose[0],
+                _nav_target_pose[1],
+                np.rad2deg(_nav_target_pose[2]),
+            )
+            check_navigation_suc = is_pose_within_bounds(
+                result[0][-1].get("pose"),
+                _nav_target_pose_deg,
+                self.nav_config.SUCCESS_DISTANCE,
+                self.nav_config.SUCCESS_ANGLE_DIST,
+            )
+            return check_navigation_suc, "Successfully reached the target pose"
+        elif not self._should_record_trajectories:
+            return True, "Successfully reached the target pose by default"
```

Comment:
Thank you for fixing the bug Jimmy, looks good!

Did you test this on spot too?

Reply:
Yes, I did! See the video on the slack channel!